src/app/task/task-list/task-list.component.html

```diff
@@ -2,9 +2,9 @@
   <div class="list-container">
     <h2>Task List</h2>
     <div class="task-list">
-      @for (task of taskList; track $index; let last = $last) {
+      @for (task of taskList; track task.id; let last = $last) {
         <div class="list-item">
-          <task-list-item [task]=task [index]=$index/>
+          <task-list-item [task]=task (updateListEvent)="ngOnInit()"/>
```

Comment:
updateListEvent should trigger an updateTask method in task-list component, where you can then call the taskService.updateTask like you do in the task-list-item component. ngOnInit shouldn't be used here only in the component.ts